### DISCUSSION TITLE
runServices and compose depends_on

### DISCUSSION THREAD

**Discussion by dtheodor** [devcontainers/spec#274] [General]
The `runServices` devcontainer property for docker compose specifies additional services from the compose file that should be started. If not specified, it starts all of them.

But doesn't the `depends_on` setting in compose itself cover this need? With the much saner default of not starting any additional services if not specified. Is there any rationale that I am missing here? At the moment it feels like I just need to specify these twice for no good reason.

**Comment by slominskir** [comment]
There is definitely overlap and confusion here.  To make matters worse, the IntelliJ (JetBrains) devcontainer runtime behaves differently here than the Microsoft codespaces / VSCode devcontainer runtime.   In the absence of runServices Microsoft's products effectively invoke `docker compose up` whereas JetBrains invokes `docker compose up <dev-container-name>`.   And without depends_on that means JetBrains defaults to launching only the devcontainer itself and no other services defined in the compose.yaml whereas Microsoft defaults to launching all services in this same scenario.
